src/components/input-components.js

```diff
@@ -0,0 +1,33 @@
+// standard class with all states found in the other inputs of this work
+
+export default class Input {
+    constructor(required = true) {
+        this.disabled = false;
+        this.required = required;
+    }
+
+    enable() {
+        this._textInput.prop('disabled', false);
+        this.disabled = this._textInput.is(':disabled');
+    }
+
+    disable() {
+        this._textInput.prop('disabled', true);
+        this.disabled = this._textInput.is(':disabled');
+    }
+
+    setRequired(boolean) {
+        this._textInput.prop('required', boolean);
+        this.required = boolean;
+    }
+
+    isRequired() {
```

Comment:
I will know if it is required by accessing the property and not by the method

Reply:
fixed and remove this method.